js/src/errors.js

```diff
@@ -3,42 +3,50 @@
  * @author Nicholas C. Zakas
  */
 
+//-----------------------------------------------------------------------------
+// Typedefs
+//-----------------------------------------------------------------------------
+
+/** @typedef {import("./momoa").MomoaLocation} MomoaLocation */
+/** @typedef {import("./momoa").MomoaToken} MomoaToken */
+
+//-----------------------------------------------------------------------------
+// Errors
+//-----------------------------------------------------------------------------
 
 /**
  * Base class that attaches location to an error.
  */
 export class ErrorWithLocation extends Error {
 
     /**
-     * 
+     * Creates a new instance.
      * @param {string} message The error message to report. 
-     * @param {int} loc.line The line on which the error occurred.
-     * @param {int} loc.column The column in the line where the error occurrred.
-     * @param {int} loc.index The index in the string where the error occurred.
+     * @param {MomoaLocation} loc The location information for the error.
      */
-    constructor(message, { line, column, index }) {
+    constructor(message, { line, column, offset }) {
         super(`${ message } (${ line }:${ column})`);
 
         /**
          * The line on which the error occurred.
-         * @type int
+         * @type number
          * @property line
          */
         this.line = line;
 
         /**
          * The column on which the error occurred.
-         * @type int
+         * @type number
          * @property column
          */
         this.column = column;
         
         /**
          * The index into the string where the error occurred.
-         * @type int
-         * @property index
+         * @type number
+         * @property offset
```

Comment:
`@property` isn't needed by typescript to identify instance properties (but might be needed by a documentation generator if you use one)